Framework/PythonInterface/mantid/fitfunctions.py

```diff
@@ -1,377 +1,386 @@
-  def __init__ (self, name, **kwargs):
-    """ 
-    Called when creating an instance
-    :param name: name of fitting function to create or 
-    an Ifunction object to wrap.
-    :param **kwargs: standard argument for __init__ function
+    """ Wrapper class for Fitting Function
     """
-    if not isinstance(name, str):
-       self.fun = name
-    else:
-       self.fun = FunctionFactory.createFunction(name)
-       # Deal with attributes first
-       for key in kwargs:
-          if key == "attributes":
-             atts = kwargs[key]
-             for keya in atts:
-               self.fun.setAttributeValue(keya, atts[keya])
-          elif self.fun.hasAttribute(key):
-             self.fun.setAttributeValue(key, kwargs[key])
-           
-       # Then deal with parameters        
-       for key in kwargs:
-          if key != "attributes" and not self.fun.hasAttribute(key):
-             self.fun.setParameter(key, kwargs[key])
- 
-  def __getattr__(self, item):
-      """
-      __getattr__ invoked when attribute item not found in the instance,
-      nor in the class, nor its superclasses.
-      :param item: named attribute
-      :return: attribute of self.fun instance, or any of the fitting
-       parameters or function attributes
-      """
-      if 'fun' in self.__dict__:
-          if hasattr(self.fun, item):
-              return getattr(self.fun, item)
-          else:
-              return self.__getitem__(item)
- 
-  def __setattr__(self, key, value):
-      """
-      __setattr__ invoked when attribute key not found in the instance,
-      nor in the class, nor its superclasses.
-      Enables direct access to the attributes of self.fun instance, and
-      also to its fitting parameters and function attributes
-      :param key: named attribute 
-      :param value: new value for the named attribute
-      """
-      if key == 'fun':
-          self.__dict__['fun'] = value  # initialize self.fun
-      elif 'fun' in self.__dict__ and hasattr(self.fun, key):
-          setattr(self.fun, key, value)  # key is attribute of instance self.fun
-      elif 'fun' in self.__dict__ and self.fun.hasAttribute(key):
-          self.fun.setAttributeValue(key, value)  # key is function attribute
-      elif 'fun' in self.__dict__ and self.fun.hasParameter(key):
-          self.fun.setParameter(key, value)  # key is fitting parameter
-      else:
-          self.__dict__[key] = value  # initialize self.key
+    def __init__ (self, name, **kwargs):
+        """
+        Called when creating an instance
+        :param name: name of fitting function to create or
+        an Ifunction object to wrap.
+        :param **kwargs: standard argument for __init__ function
+        """
+        if not isinstance(name, str):
+            self.fun = name
+        else:
+            self.fun = FunctionFactory.createFunction(name)
+        self.init_paramgeters_and_attributes(**kwargs)
+
+    def init_paramgeters_and_attributes(self, **kwargs):
```

Comment:
is `paramgeters` the intended word in method `init_paramgeters_and_attributes` ?